File: 004 - Largest palindrome product/4.c
#include <stdio.h>

int main (void){

    int i,j,k,m;

    for(i=9; i>0; i--){
        for(j=9; j>=0; j--){
            for(k=9; k>=0; k--){

                int a = (i*100001)+(j*10010)+(k*1100);

                for(m=999; m>=100; m--){

                    if(a%m==0){
                        int d = a/m;

                        if(d>=100 && d<=999){
                            printf("Largest palindrome with 3 digit multipliers is : %d\n", a);
                            return 1;
                        }

                    }

                }
            }
        }
    }

}

File: project_euler/004 - Largest palindrome product/4.py
def main():
    for i in range(9, 0, -1):
        for j in range(9, -1, -1):
            for k in range(9, -1, -1):
                a = (i*100001)+(j*10010)+(k*1100)

                for m in range(999, 101, -1):
                    if a%m==0:
                        d = a/m
                        if (d>=100 and d<=999):
                            print "Largest palindrome with 3 digit multipliers is: ", a
                            return True

main()
                        
